src/envelopeCommands.cpp

```diff
@@ -194,26 +192,43 @@ void moveToEnvelopePoint(int direction, bool clearSelection=true) {
 	if (direction != 0 && direction == 1 ? time < now : time > now)
 		return; // No point in this direction.
 	fakeFocus = FOCUS_ENVELOPE;
-	if (clearSelection)
+	currentEnvelopePoint = point;
+	if (clearSelection) {
 		Main_OnCommand(40331, 0); // Envelope: Unselect all points
-	SetEnvelopePointEx(envelope, currentAutomationItem, point, NULL, NULL, NULL, NULL, &bTrue, &bTrue);
+		isSelectionContiguous = true;
+	}
+	if(select)
+		SetEnvelopePointEx(envelope, currentAutomationItem, point, NULL, NULL, NULL, NULL, &bTrue, &bTrue);
 	if (direction != 0)
 		SetEditCurPos(time, true, true);
 	ostringstream s;
 	s << "point " << point + 1 << " value ";
 	char out[64];
 	Envelope_FormatValue(envelope, value, out, sizeof(out));
 	s << out;
-	if (!clearSelection) {
+	bool isSelected;
+	GetEnvelopePointEx(envelope, currentAutomationItem, point, NULL, NULL, NULL, NULL, &isSelected);
+	if (isSelected) {
 		int numSel = countSelectedEnvelopePoints(envelope, true);
 		// One selected point is the norm, so don't report selected in this case.
 		if (numSel > 1)
 			s << " selected";
+	} else {
+		s << " unselected ";
 	}
 	s << " " << formatCursorPosition();
 	outputMessage(s);
 }
 
+bool toggleCurrentEnvelopePointSelection() {
+	TrackEnvelope* envelope = GetSelectedEnvelope(0);
+	bool isSelected;
+	GetEnvelopePointEx(envelope, currentAutomationItem, currentEnvelopePoint, NULL, NULL, NULL, NULL, &isSelected);
```

Comment:
There's no guarantee that currentEnvelopePoint still exists. We should validate that and fail if it doesn't.